3761/1/top_gems.rb

```diff
@@ -1,29 +1,27 @@
-require './parser_argv.rb'
-require './initial_list_of_gems.rb'
+require './argv_parser.rb'
+require './gem_list_creator.rb'
 require './page.rb'
 require 'mechanize'
-require './table.rb'
-require './gem_statisic_fetcher.rb'
-parser = ParserArgv.new(ARGV)
-file_name = parser.file_name
-file_name ||= '.top-gems.yml'
-gems = InitialListOfGems.new(file_name)
-list_gems = if parser.part_name_gem.nil?
-              gems.list_gems
-            else
-              gems.sort(parser.part_name_gem)
-            end
+require 'terminal-table'
+require './gem_statistic_fetcher.rb'
+require './gem_statistic_analyzer.rb'
+require './gem_serializer.rb'
+parser = ArgvParser.new(ARGV)
```

Comment:
Move all this logic to class and call class method

Reply:
What's logic?

Reply:
Answer the question: Which class is `parser` var? and all the code bellow
It's better to move all this logic to class and then just call something like `TopGem.start/call/perform/<as_u_wish>`
outside of class for calling it when you run your CLI